src/test/java/uk/ac/ebi/embl/template/TemplateEntryProcessorTest.java

```diff
@@ -0,0 +1,170 @@
+
+        SourceFeature sourceFeature=templateProcessorResultSet.getEntry().getPrimarySourceFeature();
+        assertTrue(templateProcessorResultSet.getValidationResult().isValid());
+        assertEquals(sourceFeature.getTaxId(),Long.valueOf(taxId));
+        assertEquals(sourceFeature.getScientificName(),"Homo sapiens");
+    }
+    
+    public void executeEntryProcessInvalidOrganism(String scientificName, TemplateInfo templateInfo, String molType) throws Exception{
+
+        TemplateVariables templateVariables = getTemplateVariables_ERT000002(scientificName);
+        TemplateProcessorResultSet templateProcessorResultSet = templateEntryProcessor.processEntry(templateInfo, molType, templateVariables,getOptions().getProjectId());
+
+        Collection<ValidationMessage<Origin>> messages=templateProcessorResultSet.getValidationResult().getMessages();
+        assertFalse(templateProcessorResultSet.getValidationResult().isValid());
+    }
+    
+    private TemplateEntryProcessor getTemplateEntryProcessor() {
+        Connection con=null;
+        return new TemplateEntryProcessor(con);
+    } 
+    
+    private TemplateVariables getTemplateVariables_ERT000002(String organismName){
+        Map<String, String> variablesMap=new HashMap<>();
+        variablesMap.put("ORGANISM_NAME",organismName);
+        variablesMap.put("SEQUENCE","ACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACATACTACGACAT");
+        variablesMap.put("SEDIMENT","5S");
+        variablesMap.put("ENV_SAMPLE","no");
+        TemplateVariables templateVariables=new TemplateVariables(1,variablesMap);
+        return templateVariables;
+    }
+    
+    private SubmissionOptions getOptions(){
+        SubmissionOptions options=new SubmissionOptions();
+        options.webinCliTestMode=true;
+        options.webinAuthToken = Optional.of(getAuthToken());
+        return options;
+    }
+    
+    private String getAuthToken(){
+
+        if(StringUtils.isNotEmpty(token)){
+            return token;
+        }
+        RestTemplate restTemplate=new RestTemplate();
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        HttpEntity<String> request =
+                new HttpEntity<String>(AUTH_JSON, headers);
+        ResponseEntity<String> response =
+                restTemplate.postForEntity(TEST_AUTH_URL,request, String.class);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        token = response.getBody();
+        return token;
+    }
+    
+    private String getMolTypeFromTemplateForTest(TemplateInfo templateInfo) throws Exception {
```

Comment:
- Just a general comment: prefer runtime exceptions everywhere else than in legacy code where we already (unfortunately) use checked exceptions.